### ISSUE TITLE
Interactive learning throw an exception while export & quit sometimes

### ISSUE THREAD

**Issue by grapecitycarlzhang** [RasaHQ/rasa#4143]
Hi,

I use rasa hosting by docker(iin powershell on win10) to interactive learning,
when I finish the interactive learning and choose export & quit option, the error occurs sometimes.
```
docker run -it --rm -v ${pwd}:/app rasa/rasa:1.1.4-full interactive
```
```
? Do you want to stop?  Export & Quit
? Export stories to (if file exists, this will append the stories) data/stories.md
? Export NLU data to (if file exists, this will merge learned data with previous training examples) data/nlu.md
? Export domain file to (if file exists, this will be overwritten) domain.yml
2019-07-30 11:00:02 INFO     rasa.nlu.training_data.loading  - Training data format of data/nlu.md is md
2019-07-30 11:00:02 INFO     rasa.nlu.training_data.training_data  - Training data stats:
        - intent examples: 59 (11 distinct intents)
        - Found intents: 'greet', 'start_chart', 'say_shhh', 'goodbye', 'describe_image_bed', 'deny', 'say_wake_up', 'describe_image_sleep', 'mood_unhappy', 'mood_great', 'affirm'
        - entity examples: 0 (0 distinct entities)
        - found entities:

2019-07-30 11:00:02 INFO     rasa.nlu.training_data.training_data  - Training data stats:
        - intent examples: 4 (2 distinct intents)
        - Found intents: 'out_of_scope', 'start_chart'
        - entity examples: 0 (0 distinct entities)
        - found entities:

2019-07-30 11:00:02 INFO     rasa.nlu.training_data.training_data  - Training data stats:
        - intent examples: 63 (12 distinct intents)
        - Found intents: 'greet', 'start_chart', 'say_shhh', 'out_of_scope', 'goodbye', 'describe_image_bed', 'deny', 'say_wake_up', 'describe_image_sleep', 'mood_unhappy', 'mood_great', 'affirm'
        - entity examples: 0 (0 distinct entities)
        - found entities:

2019-07-30 11:00:02 ERROR    rasa.core.training.interactive  - An exception occurred while recording messages.
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 1360, in record_messages
    await _enter_user_message(sender_id, endpoint)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 1199, in _enter_user_message
    message = await _ask_questions(question, sender_id, endpoint, lambda a: not a)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 312, in _ask_questions
    should_retry = await _ask_if_quit(sender_id, endpoint)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 589, in _ask_if_quit
    await _write_data_to_file(sender_id, endpoint)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 565, in _write_data_to_file
    await _write_nlu_to_file(nlu_path, events)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 807, in _write_nlu_to_file
    f.write(nlu_data.as_markdown())
  File "/usr/local/lib/python3.6/site-packages/rasa/nlu/training_data/training_data.py", line 133, in as_markdown
    return MarkdownWriter().dumps(self)
  File "/usr/local/lib/python3.6/site-packages/rasa/nlu/training_data/formats/markdown.py", line 206, in dumps
    md += self._generate_training_examples_md(training_data)
  File "/usr/local/lib/python3.6/site-packages/rasa/nlu/training_data/formats/markdown.py", line 217, in _generate_training_examples_md
    key=lambda k: k["intent"],
  File "/usr/local/lib/python3.6/site-packages/rasa/nlu/training_data/formats/markdown.py", line 217, in <lambda>
    key=lambda k: k["intent"],
KeyError: 'intent'
2019-07-30 11:00:02 ERROR    asyncio  - Task exception was never retrieved
future: <Task finished coro=<_serve_application.<locals>.run_interactive_io() done, defined at /usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py:1405> exception=KeyError('intent',)>
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 1412, in run_interactive_io
    sender_id=uuid.uuid4().hex,
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 1360, in record_messages
    await _enter_user_message(sender_id, endpoint)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 1199, in _enter_user_message
    message = await _ask_questions(question, sender_id, endpoint, lambda a: not a)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 312, in _ask_questions
    should_retry = await _ask_if_quit(sender_id, endpoint)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 589, in _ask_if_quit
    await _write_data_to_file(sender_id, endpoint)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 565, in _write_data_to_file
    await _write_nlu_to_file(nlu_path, events)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/training/interactive.py", line 807, in _write_nlu_to_file
    f.write(nlu_data.as_markdown())
  File "/usr/local/lib/python3.6/site-packages/rasa/nlu/training_data/training_data.py", line 133, in as_markdown
    return MarkdownWriter().dumps(self)
  File "/usr/local/lib/python3.6/site-packages/rasa/nlu/training_data/formats/markdown.py", line 206, in dumps
    md += self._generate_training_examples_md(training_data)
  File "/usr/local/lib/python3.6/site-packages/rasa/nlu/training_data/formats/markdown.py", line 217, in _generate_training_examples_md
    key=lambda k: k["intent"],
  File "/usr/local/lib/python3.6/site-packages/rasa/nlu/training_data/formats/markdown.py", line 217, in <lambda>
    key=lambda k: k["intent"],
KeyError: 'intent'
```

**Comment by msamogh** [comment]
Thanks for raising this issue, @MetcalfeTom will get back to you about it soon.

**Comment by tmbo** [comment]
@grapecitycarlzhang Can you please attach the initial training data as well as the whole log file? (so we can see what you entered and selected) this would make it a lot easier to reproduce

**Comment by MetcalfeTom** [comment]
Hi @grapecitycarlzhang,

This issue was [fixed in 1.1.8](https://github.com/RasaHQ/rasa/pull/3976) so please update and let us know if the error persists.

**Comment by grapecitycarlzhang** [comment]
Thanks,
I have never see the issue when upgrade to 1.1.8.
